### PR TITLE
switch to NCCL 2.8.3 built from source for CuPy, Horovod, libgpuarray, PyTorch and TensorFlow using fosscuda/2020b (+ add missing astor extension for TensorFlow)

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

edit: requires ~#13071~ (NCCL 2.8.3 from source)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 18 out of 21 (8 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/e9a292f37996f8d77d9d86a733335d92 for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb [hunk 1]
```diff
@@ -119,8 +119,8 @@ exts_list = [
         'unpack_sources': False,
         'checksums': ['b75e034300ccb169403cf2695adf3368da68863aeb0c14c3760064c713d5c486'],
     }),
-    ('astor', '0.8.0', {
-        'checksums': ['37a6eed8b371f1228db08234ed7f6cfdc7817a3ed3824797e20cbb11dc2a7862'],
+    ('astor', '0.8.1', {
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb:122]: @Flamefire Is there a specific need for the version bump? If not, should we avoid the in-place update?

Comment (Flamefire) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb:122]: Yes: it needs the bugfix for the new setuptools we now use. See @Micket report in Slack and https://github.com/berkerpeksag/astor/commit/2159cfeb905af82eaf0cdf6beaa5e7b486a71c86

TLDR: Us correctly installing setuptools now requires the new astor

**Review by Micket — APPROVED** [review]
lgtm (i hope to have a test report coming in after.. many hours)

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb:122]: I think this is fine to change: Those who have built the old version from before can just continue to use that, anyone building a new one now would fail, so we should fix this.

Comment (Flamefire) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb:122]: Yeah, see my report above: https://gist.github.com/Flamefire/b9c61028995b9e5f75e992325bb470db

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/80508e6a75fd63dcca52f793c893b9d0 for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-fosscuda-2020a-Python-3.8.2.eb:122]: OK, understood, thanks for clarifying!

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/1e677b516a1ebef6b6864c714eb01711 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/23a93cda074a3c06a75c19f69ed1a973 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/de39519e6390888866d1837df08a7aa5 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (2 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/33ea69f67ca08bdcfe92576ea17705b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (8 easyconfigs in total)
node3307.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/2fb7767f4d1384ec7fb54c6ae161ce2d for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/8a1260556e32023ab135bc8d527d0b53 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
alvis1-02 - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, Python 3.6.8
See https://gist.github.com/a239d2b379ede6c1f81f37a610d871d5 for a full test report.
